### PR TITLE
Make DM map and command functions to outside of ServiceProxy.

### PR DESCRIPTION
**Pull request by ehpor**
This allows us to operate on DM commands without spinning up an entire simulator.

### CODE REVIEW

File: catkit2/testbed/proxies/deformable_mirror.py [hunk 1]
```diff
@@ -148,20 +138,80 @@ def write_dm_shape(self, fname, dm_shape):
...
+        The path where to write the DM shape.
+    dm_shape : ndarray
+        Either a DM command or DM maps.
+    device_actuator_mask : array
+        A mask describing the physical actuators on all DMs.
+    '''
+    command_length = np.sum(device_actuator_mask)
+
+    # If we're given a DM map, convert it to a command.
+    if dm_shape.shape != (command_length,):
```
Comment (ivalaginja) [inline catkit2/testbed/proxies/deformable_mirror.py:202]: Is the comma here on purpose?

Comment (ehpor) [reply catkit2/testbed/proxies/deformable_mirror.py:202]: Yes, it's a comparison with a tuple. No new code, it was just moved from inside to outside the class.

Comment (ivalaginja) [reply catkit2/testbed/proxies/deformable_mirror.py:202]: oh right

**Review by ivalaginja — APPROVED** [review]
I had hacked my way through this for THD2, thanks for actually doing this properly.

**Pull request merged**